packages/xo-server/src/xo-mixins/pool.js

```diff
@@ -0,0 +1,61 @@
+import { identity, differenceBy } from 'lodash'
+import { mapToArray } from '../utils'
+
+export default class {
+  constructor(xo) {
+    this._xo = xo
+  }
+
+  async merge({ source, target, force }) {
```

Comment:
Let's ask @julien-f where this should actually go :)

So we're merging 1 or more hosts in a pool, so we need to:
1- compare the installed patches on all the hosts
2- install the missing ones on all the "source" hosts
3- handle the servers registrations and the XAPI instances
4- call the `pool.join`s

We have:
- "add one host" and "add multiple hosts" in api/pool (← we can probably only have one without even breaking the API btw)
- `xo-mixins/xen-servers#mergeXenPools` that handles (3) and (4) for 1 host

So where do you think the patches handling should go? We're trying to remove it from the API and it can't go on the XAPI side since we're comparing 2 pools. So is a xo-mixin ok?

Reply:
Yes, IMO it makes sense in a mixin :slightly_smiling_face: